distributionviewer/core/static/css/distributionviewer.styl

```diff
@@ -156,3 +156,49 @@ a {
     width: 100%;
   }
 }
+
+/* Fetching Animation Component */
+.is-fetching {
+  background-color: $background-color;
+  min-height: 100px;
+  position: relative;
+
+  > * {
+    display: none;
+  }
+  .fetching {
+    display: block;
+  }
+}
+
+.fetching {
+  display: none;
+  font-family: consolas, menlo, monaco, monospace;
+  font-size: 4rem;
+  font-weight: bold;
+  height: 100%;
+  left: 50%;
+  margin: -30px 0 0 -20px;
+  opacity: 0.8;
+  position: absolute;
+  top: 50%;
+  width: 100%;
+
+  span {
+    display: inline-block;
+    animation: pulse 0.4s alternate infinite ease-in-out;
+
+    &:nth-child(odd) {
+      animation-delay: 0.4s;
+    }
+  }
+}
+
+@keyframes pulse {
+  to {
+    transform: scale(0.8);
+    opacity: 0.5;
+  }
+}
+/* End Fetching Animation Component */
```

Comment:
If we wanted to, we could even organize this code further by putting it into a different file. The gulp task will pick up on it. Just a question of personal preference really. Up to you.